app/views/unattended/provisioning_templates/finish/windows_default_finish.erb

```diff
@@ -52,15 +57,22 @@ params:
   w32tm /resync
   w32tm /resync
 
-  :: You can join your machine to the domain right here >
-
-  <% if host_param('domainAdminAccount') and host_param('domainAdminAccountPasswd') -%>
-    echo joining domain
-    powershell.exe -OutputFormat text -ExecutionPolicy remotesigned -command c:\deploy\joinDomain.ps1
+  <% if host_param('computerDomain') -%>
+    <% if host_param('domainAdminAccount').present? and host_param('domainAdminAccountPasswd').present? -%>
+      echo performing secure domain join
+      echo $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList ([pscustomobject]@{ UserName = '<%= host_param('domainAdminAccount') -%>'; Password = (ConvertTo-SecureString -String '<%= host_param('domainAdminAccountPasswd') -%>' -AsPlainText -Force)[0] }) > c:\deploy\joinDomain.ps1
+      echo Add-Computer -DomainName '<%= host_param('computerDomain') -%>' -Credential $Credential <% if host_param('computerOU').present? -%>-OUPath '<%= host_param('computerOU') -%>'<% end -%> >> c:\deploy\joinDomain.ps1
+      powershell.exe -OutputFormat text -command c:\deploy\joinDomain.ps1
+    <% else %>
+      <% if host_param('machinePassword').present? %>
+        echo performing unsecure domain join
+        echo $Credential = New-Object pscredential -ArgumentList ([pscustomobject]@{ UserName = $null; Password = (ConvertTo-SecureString -String '<%= host_param('machinePassword') -%>' -AsPlainText -Force)[0] })  > c:\deploy\joinDomain.ps1
+        echo Add-Computer -Domain '<%= host_param('computerDomain') -%>' -Options UnsecuredJoin,PasswordPass -Credential $Credential >> c:\deploy\joinDomain.ps1
+        powershell.exe -OutputFormat text -command c:\deploy\joinDomain.ps1
```

Comment:
Just wondering if this could not be simply something like "powershell New-Object pscredential ...". Like calling a `bash -c "command". But I am good with this too, I have zero understanding of powershell.

Reply:
had to do some testing, this works indeed also